Build/scripts/pre_build.ps1

```diff
@@ -162,10 +162,38 @@ $CommitMessage
     $changeJson | Add-Member -type NoteProperty -name Username -value $Env:Username
     $changeJson | Add-Member -type NoteProperty -name CommitMessage -value $CommitMessageLines
 
+    ## Look for a git note that requests a cirro run.
+
+    ## We assume that, when this script is called in the context of a full build, the working directory
+    ## is the root of the local full checkout.
+
+    ## It's a bit of a layering violation to have this logic, which only applies to full builds, in the
+    ## core pre_build script, but there's not an obviously better place to put it.  There is a pre_build
+    ## script in the full repo, but it doesn't seem to be executed as part of a VSO build, and it's not
+    ## clear if anyone else uses it.
+    ##
+    ## Since we have to handle the case where a full build doesn't have a ChakraHub note anyway, just
+    ## add the logic to the core script.
+
+    git -c http.extraheader="AUTHORIZATION: bearer ${Env:System.AccessToken}" fetch origin refs/notes/ChakraHub:refs/notes/ChakraHub
```

Comment:
Might be worth adding a check to see if `${Env:System.AccessToken}` exists? That way we can choose not to expose it to core builds if it's irrelevant (and we'll save the git fetch overhead)

Reply:
Yes, that occurred to me last night; I'll add the change.  Since we're fetching only a specific ref that we expect to be quite small, the fetch operation doesn't take very long, but you're right, it would be good to avoid it when possible.